corehq/motech/repeaters/models.py

```diff
@@ -1038,6 +1039,14 @@ def _migration_get_submodels(cls):
     def _migration_get_sql_model_class(cls):
         return SQLRepeatRecord
 
+    def save(self, *args, sync_attempts=False, **kw):
+        with enable_attempts_sync_to_sql(self, sync_attempts):
+            return super().save(*args, **kw)
+
+    def _migration_sync_submodels_to_sql(self, sql_object):
+        if self._should_sync_attempts:
```

Comment:
I didn't follow the paths of what calls `__migration_sync_to_sql()`, but I'm guessing that you are sure that this cannot be reached outside the `enable_attempts_sync_to_sql()` context manager?

Reply:
Yes, this is only called within the context of `.save()`, which uses the context manager. Having the attribute not set outside the context manager will alert us (hopefully in tests, if not then in Sentry) if any code path arises that violates that precondition.